### PR TITLE
issue-211: support v.2.0.0 for delete-ltp-and-dependents

### LINKED ISSUE #211: Support Version 2.0.0 changes for delete-ltp-and-dependents
1. Generate server stub for the version 2.0.0 specification 
2. If the api/openapi.yaml is not updated as per the published version 2.0.0 , then copy the api/openapi.yaml file from the generated stub to the source code
3. no change required in the method signature

- [x] In the version 1 implementation , the information included in the configuration file will be deleted. But , in the version 2 , since the topology information of other applications will be available in ElasticSearch database , the refered ltp will be deleted from the application data (i.e) elastic search. [ElasticSearch://core-model-1-4:network-control-domain/control-construct=*/logical-termination-point=*/]
- [x] If a serving HttpClient gets deleted, all client OperationClients get deleted too.
- [x] If the last serving TcpClient gets deleted, the client HttpClient gets deleted too.
- [x] If an OperationClient is referenced in an FcPort or a LinkPort, the referencing FcPort or LinkPort gets deleted, too.

### PR DESCRIPTION
**Pull request by DanaSunal**
Fixes #211

### CODE REVIEW

**Review by vanithavalluripalli9 — CHANGES_REQUESTED** [review]

File: server/service/individualServices/LinkServices.js [hunk 1]
```diff
@@ -105,24 +107,49 @@ exports.findOrCreateLinkForTheEndPointsAsync = function (EndPoints) {
...
- * @param {String} linkUuid : uuid of the link
- * @param {String} consumingOperationuuid : logical-termination-point of the link-port
- * @return {Promise} boolean {true|false}
+ * @description This function deletes a link from Elasticsearch
+ * @param {String} linkId : documentId for the link
+ * @return {Promise<Object>} Elastincsearch response
  **/
- function deleteLinkAsync(linkUuid,consumingOperationuuid) {
-    return new Promise(async function (resolve, reject) {
-        try {
```
[the remark this answers is not in the document]
Comment (DanaSunal) [reply server/service/individualServices/LinkServices.js:115]: The service remove-operation-client-to-link should also use the new method from this file (that is correctly called deleteLinkPortAsync - line 132)

[the remark this answers is not in the document]
Comment (vanithavalluripalli9) [reply server/service/individualServices/LinkServices.js:115]: Thank you for indicating . Yeah, it is indeed using the correct method deleteLinkPortAsync and pull request #255 also has the same method contents . The method deleteLinkPortAsync is being called from deleteLinkAsync

File: server/service/individualServices/LinkServices.js [hunk 2]
```diff
@@ -339,4 +367,35 @@ function getOperationClientUuid(controlConstruct, operationClientName, consuming
...
+    if (Object.keys(res.body).length === 0) {
+      return;
+    }
+    let linkPorts = res.body.hits.hits[0]._source[onfAttributes.LINK.LINK_PORT];
+    let found = linkPorts.find(linkPort => linkPort[onfAttributes.LINK.LOGICAL_TERMINATION_POINT] === uuid);
+    let linkUuid = res.body.hits.hits[0]._source[onfAttributes.GLOBAL_CLASS.UUID];
+    if (LinkPort.portDirectionEnum.INPUT === found[onfAttributes.LINK.PORT_DIRECTION]) {
+      deleteLinkPortAsync(linkUuid, found[onfAttributes.LOCAL_CLASS.LOCAL_ID]);
+    } else {
+      deleteLinkAsync(res.body.hits.hits[0]._id);
```
Comment (vanithavalluripalli9) [inline server/service/individualServices/LinkServices.js:398]: For now there is no scenario where we will be deleting the whole link . so we can remove this

**Review by vanithavalluripalli9 — DISMISSED** [review]

File: server/service/individualServices/LinkServices.js [hunk 3]
```diff
@@ -105,24 +107,51 @@ exports.findOrCreateLinkForTheEndPointsAsync = function (EndPoints) {
...
+ * @return {Promise<Object>} Elastincsearch response
  **/
- function deleteLinkAsync(linkUuid,consumingOperationuuid) {
-    return new Promise(async function (resolve, reject) {
-        try {
-            let localId = await Link.getLocalIdOfTheConsumingOperationAsync(linkUuid,consumingOperationuuid)
-            if (localId) {
-                await Link.deleteLinkPortAsync(linkUuid,localId);
+ async function deleteLinkAsync(linkId) {
+  let esUuid = await ElasticsearchPreparation.getCorrectEsUuid(true);
```
Comment (vanithavalluripalli9) [inline server/service/individualServices/LinkServices.js:115]: As the method is not used deleteLinkAsync(linkId) anywhere can we remove it and restore the older one of deleteLinkAsync(linkUuid , consumingOperationUuid )

**Review by vanithavalluripalli9 — CHANGES_REQUESTED** [review]

File: server/service/individualServices/LinkServices.js (hunk 3 above)
Comment (DanaSunal) [reply server/service/individualServices/LinkServices.js]: Done.

**Review by vanithavalluripalli9 — APPROVED** [review]

**Pull request merged**